### PR TITLE
4094 openjdk11

### LINKED ISSUE #4094: Technology upgrade - Java / Eclipse / Others
This issue covers the steps to upgrade the technology we're sitting on.

One customer base was reliant on Java7.  But, now that they're happy to take an embedded JRE we can upgrade.

Alongside this, we will also upgrade some library dependencies.

Ian _believes_ this is the sequence:
* [x] update build to Java 8
* [x] update Eclipse to 2019-03
* [ ] update build to Java 11 (not Java 9, since it's already past end-of-life)
* [ ] update GeoTools to newest stable release
* [ ] update Radiance/Flamingo to newest stable release (#4562 )

Note: there some GeoTools build tips here:
https://docs.geotools.org/latest/userguide/build/install/jdk.html#building-geotools-with-java-11

Here is the upgrade path, through a series of GeoTools versions. Significantly, it includes how to handle Units of Measurement issues:
https://docs.geotools.org/latest/userguide/welcome/upgrade.html

### PR DESCRIPTION
**Pull request by theanuradha**
Fixes #4094

### CODE REVIEW

File: org.mwc.debrief.dis/src/org/mwc/debrief/dis/runner/SimulationRunner.java [hunk 1]
```diff
@@ -151,7 +151,7 @@ public void stop()
     }
     if (oThread != null)
     {
-      oThread.destroy();
+     // oThread.destroy();
```
Comment (IanMayo) [inline org.mwc.debrief.dis/src/org/mwc/debrief/dis/runner/SimulationRunner.java]: @theanuradha - do you remember why you had to comment out this line?

Comment (IanMayo) [reply org.mwc.debrief.dis/src/org/mwc/debrief/dis/runner/SimulationRunner.java]: Aaah, don't worry. I've researched it, and added a comment.

File: org.mwc.debrief.lite/src/main/java/org/mwc/debrief/lite/DebriefLiteApp.java [hunk 2]
```diff
@@ -466,7 +466,8 @@ private static void setInitialArea(final JMapPane mapPane,
       Application.logError2(ToolParent.ERROR,
           "Failure in setting initial viewport coverage", e);
     }
-    mapPane.getMapContent().getViewport().setFixedBoundsOnResize(true);
+    //TODO:geo tools update ???
+    //mapPane.getMapContent().getViewport().setFixedBoundsOnResize(true);
```
Comment (IanMayo) [inline org.mwc.debrief.lite/src/main/java/org/mwc/debrief/lite/DebriefLiteApp.java]: this method was inserted, just for us, in this PR:
https://github.com/geotools/geotools/pull/2324/files

I _think_ we may need to switch to GeoTools 21.2 for it to be in the release.

File: info.limpet/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -47,10 +47,12 @@ Require-Bundle: info.limpet.libs;visibility:=reexport,
...
+ javax.xml.bind
 Bundle-Vendor: MWC
-Bundle-RequiredExecutionEnvironment: JavaSE-1.7
+Bundle-RequiredExecutionEnvironment: JavaSE-11
 Bundle-ClassPath: .,
  libs/xmlpull-1.1.3.1.jar,
  libs/xpp3_min-1.1.4c.jar,
- libs/xstream-1.4.8.jar
+ libs/xstream-1.4.8.jar,
+ libs/jsr-275-1.0-beta-2.jar
```
Comment (IanMayo) [inline info.limpet/META-INF/MANIFEST.MF]: Hmm @G - I think the it's this JSR that got replaced:
https://docs.geotools.org/latest/userguide/welcome/upgrade.html#migrate-to-jsr-363-units

Comment (theanuradha) [reply info.limpet/META-INF/MANIFEST.MF]: Let me check

Comment (theanuradha) [reply info.limpet/META-INF/MANIFEST.MF]: this going to be fun im on it

**Comment by helenayele** [comment]
Clipboard functionality throw the below exception but it has copied the image in the clipboard buffer.
### Steps :

1. Launch D-Lite
2. Open boat1.rep file
3. Go to File  menu and select Clipboard button
image

java.io.IOException: Registered service providers failed to encode BufferedImage@6894f557: type = 2 DirectColorModel: rmask=ff0000 gmask=ff00 bmask=ff amask=ff000000 IntegerInterleavedRaster: width = 626 height = 439 #Bands = 4 xOff = 0 yOff = 0 dataOffset[0] 0 to image/jpeg
	at java.desktop/sun.awt.datatransfer.DataTransferer.imageToStandardBytesImpl(DataTransferer.java:1799)
	at java.desktop/sun.awt.datatransfer.DataTransferer.imageToStandardBytes(DataTransferer.java:1713)
	at java.desktop/sun.awt.windows.WDataTransferer.imageToPlatformBytes(WDataTransferer.java:353)
	at java.desktop/sun.awt.datatransfer.DataTransferer.translateTransferable(DataTransferer.java:855)
	at java.desktop/sun.awt.windows.WDataTransferer.translateTransferable(WDataTransferer.java:220)
	at java.desktop/sun.awt.windows.WClipboard.setContentsNative(WClipboard.java:83)
	at java.desktop/sun.awt.datatransfer.SunClipboard.setContents(SunClipboard.java:105)
	at org.mwc.debrief.lite.util.ClipboardUtils.copyToClipboard(ClipboardUtils.java:85)
	at org.mwc.debrief.lite.menu.DebriefRibbonFile$CopyPlotAsPNG.actionPerformed(DebriefRibbonFile.java:76)
	at org.pushingpixels.flamingo.api.common.AbstractCommandButton.fireActionPerformed(AbstractCommandButton.java:607)
	at org.pushingpixels.flamingo.api.common.AbstractCommandButton$ActionHandler.actionPerformed(AbstractCommandButton.java:559)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
	at org.pushingpixels.flamingo.api.common.model.ActionRepeatableButtonModel.setPressed(ActionRepeatableButtonModel.java:114)
	at org.pushingpixels.flamingo.internal.ui.common.BasicCommandButtonListener.mouseReleased(BasicCommandButtonListener.java:123)
	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6632)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3342)
	at java.desktop/java.awt.Component.processEvent(Component.java:6397)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5008)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4840)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2772)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4840)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:772)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:743)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)

**Comment by IanMayo** [comment]
Ooh, @helenayele  - that's a good one :-)

**Comment by helenayele** [comment]
Short cut keys like ctrl + X - for cut  and ctrl + V for paste don't work also

**Comment by helenayele** [comment]
Yes, I am in D-Lite, Yes, they behave the same with prev version of D-Lite as well.

**Comment by helenayele** [comment]
In full debrief, 
1. Debrief Export buttons seem a little bit crowded
[Image from Gyazo](https://gyazo.com/e7c6deba687ec6334b616ddb565ec48f)
2. [Image from Gyazo](https://gyazo.com/c4fc89f3879a30c3da2ef6ec51c7ceea) - is this delete icon valid here for this success dialog?
3. When I select help for the first time from Time Controller drop-down, the help window is opened but logs this :
java.io.FileNotFoundException
	at org.eclipse.help.internal.toc.TocFileParser.parse(TocFileParser.java:47)
	at org.eclipse.help.internal.toc.TocFileProvider.getTocContributions(TocFileProvider.java:48)
	at org.eclipse.help.internal.toc.TocManager.getAndCacheTocContributions(TocManager.java:214)
	at org.eclipse.help.internal.toc.TocManager.getTocContributionsForToc(TocManager.java:202)
	at org.eclipse.help.internal.toc.TocManager.getRootTocContributions(TocManager.java:310)
	at org.eclipse.help.internal.toc.TocManager.getTocs(TocManager.java:71)
	at org.eclipse.help.internal.search.IndexingOperation.getAllDocuments(IndexingOperation.java:483)
	at org.eclipse.help.internal.search.IndexingOperation.getAddedDocuments(IndexingOperation.java:356)
	at org.eclipse.help.internal.search.IndexingOperation.execute(IndexingOperation.java:91)
	at org.eclipse.help.internal.search.LocalSearchManager.updateIndex(LocalSearchManager.java:633)
	at org.eclipse.help.internal.search.LocalSearchManager.ensureIndexUpdated(LocalSearchManager.java:607)
	at org.eclipse.help.internal.search.federated.IndexerJob.run(IndexerJob.java:35)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)

**Comment by helenayele** [comment]
When inserting track segment, it throws "Command 'org.mwc.debrief.core.creators.chartFeatures.InsertTrackSegment' fail" exception with the below stack trace:
[org.eclipse.core.commands.ExecutionException.txt](https://github.com/debrief/debrief/files/3783613/org.eclipse.core.commands.ExecutionException.txt)

[Image from Gyazo](https://gyazo.com/4a8ff017ba20a639442ff910e5281f3a)

**Comment by IanMayo** [comment]
@helenayele - for that `Track Segment` bug, I'll need some reproducible steps. Would you mind capturing it as a separate Issue please?

**Comment by helenayele** [comment]
> @helenayele - for that `Track Segment` bug, I'll need some reproducible steps. Would you mind capturing it as a separate Issue please?

Okay separately filed it at #4607

**Comment by helenayele** [comment]
#4605
#4606,
#4609

**Pull request merged**